### PR TITLE
Add a new dialog box for sending login information to the server

### PR DESCRIPTION
**Pull request by MangosArentLiterature**
This PR adds LoginDialog, a dialog box for inputting a username and password, and sending it to the server, removing the need to use sever commands (and thus, removing the potential to accidentally leak moderator passwords). The login dialog will send an AUTH packet to the server, containing the username and password. (If a server doesn't support usernames, the username field can be left empty).

Servers can implement support for this by adding support for the AUTH packet. The first parameter in the packet is for usernames, the second is for passwords. If your server doesn't support usernames, the first parameter in the packet can be discarded.

* Rename debug_functions to dialog_functions
* Add LoginDialog
* Add a new button to courtroom.cpp for opening the login dialog.

### CODE REVIEW

**Comment by MangosArentLiterature** [comment]
I've tested this, and it all seems to work fine. You can find an experimental branch of Akashi that supports this at https://github.com/AttorneyOnline/akashi/tree/experimental-login-changes

File: src/dialog_functions.cpp [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+LoginDialog::LoginDialog(QWidget *parent):QDialog(parent)
+{
+    setupUi();
+    setWindowTitle(QCoreApplication::translate("dialog_functions", "Login"));
+    setModal(true);
+}
+
+void LoginDialog::setupUi()
```
Comment (oldmud0) [inline src/dialog_functions.cpp:47]: For more complicated dialogs, let's use UIC, please. That way we don't have to write the UI in a C++ mess.

File: src/dialog_functions.cpp [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    editUsername = new QLineEdit(this);
+    editPassword = new QLineEdit(this);
+    editPassword->setEchoMode(QLineEdit::Password);
+
+    labelUsername = new QLabel(this);
+    labelPassword = new QLabel(this);
+    labelUsername->setText(QCoreApplication::translate("dialog_functions", "Username"));
+    labelUsername->setBuddy(editUsername);
+    labelPassword->setText(QCoreApplication::translate("dialog_functions", "Password"));
+    labelPassword->setBuddy(editPassword);
```
Comment (oldmud0) [inline src/dialog_functions.cpp:59]: Please use `tr()` instead of `QCoreApplication::translate`

File: src/courtroom.cpp [hunk 3]
```diff
@@ -5498,6 +5508,24 @@ void Courtroom::on_switch_area_music_clicked()
...
+    loginDialog->exec();
+    ui_ic_chat_message->setFocus();
+}
+
+void Courtroom::send_login_packet(QString& username, QString& password)
+{
+    QStringList packet_contents;
+    packet_contents.append(username);
+    packet_contents.append(password);
+    ao_app->send_server_packet(new AOPacket("AUTH", packet_contents));
```
Comment (oldmud0) [inline src/courtroom.cpp:5526]: This is totally convincing and secure.

**Review by oldmud0 — CHANGES_REQUESTED** [review]
I have design questions:

- This is totally convincing and secure, in the sense that it unfortunately is not. If you want to improve security, you should disable `/login` and also enhance `AUTH` to use something like challenge-response authentication.
- This needs to be behind a feature flag or otherwise it will confuse players, as tsuserver neither supports `AUTH` nor uses usernames.

**Comment by MangosArentLiterature** [comment]
This was great until I realized its actually really terrible. Marking this as a draft until I've implemented a better way of securing the communication.

**Pull request closed without merging**